process/transactions/scDeploysProcessor.go

```diff
@@ -0,0 +1,232 @@
+	if sc.selfShardID != core.MetachainShardId {
+		return nil, false
+	}
+
+	scDeploys := make([]*data.ScDeployInfo, 0)
+	for _, scr := range scrs {
+		dto := &deployerDto{sender: scr.Sender, receiver: scr.Receiver, nonce: scr.Nonce, txHash: scr.Hash, data: scr.Data}
+		deployInfo, ok := sc.searchDelegationManagerDeploy(dto, scrs)
+		if !ok {
+			continue
+		}
+
+		scDeploys = append(scDeploys, deployInfo)
+	}
+
+	if len(scDeploys) == 0 {
+		return nil, false
+	}
+
+	return scDeploys, true
+}
+
+func (sc *scDeploysProc) searchForSCRWithDeployNormal(scrs []*data.ScResult) ([]*data.ScDeployInfo, bool) {
+	scDeploys := make([]*data.ScDeployInfo, 0)
+	for _, scr := range scrs {
+		if scr.Receiver != sc.scDeployReceiverAddr {
+			continue
+		}
+		dto := &deployerDto{sender: scr.Sender, receiver: scr.Receiver, nonce: scr.Nonce, txHash: scr.Hash, data: scr.Data}
+		deployInfo, ok := sc.searchNormalDeploy(dto, scrs)
+		if !ok {
+			continue
+		}
+
+		scDeploys = append(scDeploys, deployInfo)
+	}
+
+	if len(scDeploys) == 0 {
+		return nil, false
+	}
+
+	return scDeploys, true
+}
+
+func (sc *scDeploysProc) searchNormalDeploy(dto *deployerDto, scrs []*data.ScResult) (*data.ScDeployInfo, bool) {
+	if dto.receiver != sc.scDeployReceiverAddr {
+		return nil, false
+	}
+
+	if len(scrs) < 1 {
+		return nil, false
+	}
+
+	for _, scr := range scrs {
+		if !bytes.HasPrefix(scr.Data, []byte("@6f6b")) ||
+			scr.Sender != sc.scDeployReceiverAddr ||
+			scr.Receiver != dto.sender {
+			continue
+		}
+
+		scAddress, err := sc.computeContractAddress(dto.sender, dto.nonce, factory.ArwenVirtualMachine)
+		if err != nil {
+			continue
+		}
+
+		return &data.ScDeployInfo{
+			ScAddress: scAddress,
+			TxHash:    dto.txHash,
+			Creator:   dto.sender,
+		}, true
+	}
+
+	return nil, false
+}
+
+func (sc *scDeploysProc) searchDelegationManagerDeploy(dto *deployerDto, scrs []*data.ScResult) (*data.ScDeployInfo, bool) {
+	if sc.selfShardID != core.MetachainShardId {
+		return nil, false
+	}
+
+	if dto.receiver != delegationManagerAddress {
+		return nil, false
+	}
+
+	if !bytes.HasPrefix(dto.data, []byte("createNewDelegationContract")) {
+		return nil, false
+	}
+
+	if len(scrs) < 3 {
+		return nil, false
+	}
+
+	for _, scr := range scrs {
+		if !(scr.Sender == delegationManagerAddress && scr.Receiver == dto.sender) {
+			continue
+		}
+
+		if !bytes.HasPrefix(scr.Data, []byte("@6f6b")) {
+			continue
+		}
+
+		splitData := bytes.Split(scr.Data, []byte("@"))
+		if len(splitData) < 3 {
+			continue
+		}
+
+		hexDecoded, err := hex.DecodeString(string(splitData[2]))
+		if err != nil {
+			continue
+		}
+
+		scAddress := sc.pubKeyConverter.Encode(hexDecoded)
+		return &data.ScDeployInfo{
+			ScAddress: scAddress,
+			TxHash:    dto.txHash,
+			Creator:   dto.sender,
+		}, true
+	}
+
+	return nil, false
+}
+
+func (sc *scDeploysProc) computeContractAddress(creatorAddressBech string, creatorNonce uint64, vmType []byte) (string, error) {
```

Comment:
duplicated code with the one found in elrond-go. Please instantiate a blockchain hook and use that functionality.

Reply:
created a JIRA task for this [EN-9841]  and will be done in another PR.

Reply:
great